### ISSUE TITLE
Unable to compile (error: G_ADD_PRIVATE [-Werror]) with Python 3.8

### ISSUE THREAD

**Issue by gllmhyt** [AyatanaIndicators/ayatana-ido#1]
Here's the build log when I try to rebuild the package with Python 3.8.

```
make  all-recursive
make[1]: Entering directory '/home/postblue/Téléchargements/ayatana-ido-0.4.4'
Making all in src
make[2]: Entering directory '/home/postblue/Téléchargements/ayatana-ido-0.4.4/src'
  CC       idocalendarmenuitem.lo
idocalendarmenuitem.c: In function ‘ido_calendar_menu_item_class_init’:
idocalendarmenuitem.c:87:3: warning: ‘g_type_class_add_private’ is deprecated [-Wdeprecated-declarations]
   87 |   g_type_class_add_private (gobject_class, sizeof (IdoCalendarMenuItemPrivate));
      |   ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /usr/include/gtk-3.0/gdk/gdkapplaunchcontext.h:28,
                 from /usr/include/gtk-3.0/gdk/gdk.h:32,
                 from /usr/include/gtk-3.0/gtk/gtk.h:30,
                 from idoactionhelper.h:23,
                 from idocalendarmenuitem.c:27:
/usr/include/glib-2.0/gobject/gtype.h:1308:10: note: declared here
 1308 | void     g_type_class_add_private       (gpointer                    g_class,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~
idocalendarmenuitem.c: In function ‘ido_calendar_menu_item_init’:
idocalendarmenuitem.c:109:13: error: G_ADD_PRIVATE [-Werror]
  109 |   priv = item->priv = IDO_CALENDAR_MENU_ITEM_GET_PRIVATE (item);
      |             ^~~~~~~~~~~~~~~
idocalendarmenuitem.c: In function ‘ido_calendar_menu_item_finalize’:
idocalendarmenuitem.c:139:13: error: G_ADD_PRIVATE [-Werror]
  139 |   IdoCalendarMenuItemPrivate *priv = IDO_CALENDAR_MENU_ITEM_GET_PRIVATE (item);
      |             ^~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [Makefile:673: idocalendarmenuitem.lo] Error 1
make[2]: Leaving directory '/home/postblue/Téléchargements/ayatana-ido-0.4.4/src'
make[1]: *** [Makefile:535: all-recursive] Error 1
make[1]: Leaving directory '/home/postblue/Téléchargements/ayatana-ido-0.4.4'
make: *** [Makefile:443: all] Error 2
```

**Comment by syntaxerrormmm** [comment]
I receive the same error with Python 3.7.4 on Manjaro.

**Comment by tari01** [comment]
Same thing here. Fails to build on Arch.

**Comment by tari01** [comment]
@sunweaver, please remove -Werror from line 70 in src/Makefile.am:

`-Wall -Werror -Wextra -Wno-unused-parameter -Wno-error=deprecated-declarations`

`-Wall -Wextra -Wno-unused-parameter -Wno-error=deprecated-declarations`

This fixes the build.

**Comment by sunweaver** [comment]
Hi Robert,

On  Mo 06 Jul 2020 18:10:43 CEST, Robert Tari wrote:

> @sunweaver, please remove -Werror from line 70 in src/Makefile.am:
>
> `-Wall -Werror -Wextra -Wno-unused-parameter  
> -Wno-error=deprecated-declarations`
>
> `-Wall -Wextra -Wno-unused-parameter -Wno-error=deprecated-declarations`
>
> This fixes the build.

@tari01: Are you sure that this is still an issue? I dropped the  
G_ADD_PRIVATE code with commit 0ab4079b28220f4051f8d0935a976e550514ad7f.

I just tried a build against latest Debian unstable and that went  
through just fine.

Mike
-- 

DAS-NETZWERKTEAM
c\o Technik- und Ökologiezentrum Eckernförde
Mike Gabriel, Marienthaler Str. 17, 24340 Eckernförde
mobile: +49 (1520) 1976 148
landline: +49 (4351) 850 8940

GnuPG Fingerprint: 9BFB AEE8 6C0A A5FF BF22  0782 9AF4 6B30 2577 1B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

**Comment by tari01** [comment]
> Are you sure that this is still an issue? I dropped the G_ADD_PRIVATE code with commit [0ab4079](https://github.com/AyatanaIndicators/ayatana-ido/commit/0ab4079b28220f4051f8d0935a976e550514ad7f). I just tried a build against latest Debian unstable and that went through just fine.

It was OK for some time, but now the real problem is a bunch of deprecation warnings, most notably GtkStateType vs. GtkStateFlags. There is already a PR that should make things right: #3. Removing -Werror is an extra step against warnings that crash the build.

So ideally, I think you should merge #3 and also disable -Werror

**Comment by sunweaver** [comment]
Resolved by 0ab4079b28220f4051f8d0935a976e550514ad7f.

**Comment by sunweaver** [comment]
Thus, closing...
